### ISSUE TITLE
Attributes do not travel well.

### ISSUE THREAD

**Issue by rakool** [rakudo/rakudo#4127]
## The Problem

Attributes seem to only cross certain scopes correctly.

Throw this code into a class and it works fine with or without the `get_value` line commented out.
```
    method list-attrs() {
        my $o = self;
        my $c = ::($o.^name);
        $c.^attributes.map: -> $a {
            say "{$a.name} { $a.?readonly // False ?? 'ro' !! 'rw' } "
                ~"{$a.?container.^name } "
    #           ~ "{$a.get_value( $o) } "
        }
    }
```

Move this code into a parent class and it works.

Move the parent class into a another file and it quietly  reports data from a Failure object.

```
$!exception ro Any
$!backtrace ro Any
$!handled ro int
```

Enable the `get_value` line and run with --ll-exception:
```
P6opaque: no such attribute '$!exception' on type Failure in a Child::Daughter when trying to get a value
   at SETTING::src/core.c/Attribute.pm6:180  (/priv/rir/Repo/rakudo/install/share/perl6/runtime/CORE.c.setting.moarvm:get_value)
 from /priv/rir/Repo/try/intro-o/Parent.rakumod (Parent):9  (/priv/rir/Repo/try/intro-o/.precomp/EC86CD2580B847E0847CD721DF69E986886D4662/23/23D692F07A3FB646A8E54FCD2DC724F4EA2ED2C6:)
 from /priv/rir/Repo/try/intro-o/Parent.rakumod (Parent):7  (/priv/rir/Repo/try/intro-o/.precomp/EC86CD2580B847E0847CD721DF69E986886D4662/23/23D692F07A3FB646A8E54FCD2DC724F4EA2ED2C6:)
 from SETTING::src/core.c/Any-iterable-methods.pm6:465  (/priv/rir/Repo/rakudo/install/share/perl6/runtime/CORE.c.setting.moarvm:sink-all)
 from SETTING::src/core.c/Seq.pm6:144  (/priv/rir/Repo/rakudo/install/share/perl6/runtime/CORE.c.setting.moarvm:sink)
 from attr.t:7  (<ephemeral file>:<unit>)
 from attr.t:1  (<ephemeral file>:<unit-outer>)
 from gen/moar/stage2/NQPHLL.nqp:1945  (/priv/rir/Repo/rakudo/install/share/nqp/lib/NQPHLL.moarvm:eval)
 from gen/moar/stage2/NQPHLL.nqp:2150  (/priv/rir/Repo/rakudo/install/share/nqp/lib/NQPHLL.moarvm:evalfiles)
 from gen/moar/stage2/NQPHLL.nqp:2110  (/priv/rir/Repo/rakudo/install/share/nqp/lib/NQPHLL.moarvm:command_eval)
 from gen/moar/Compiler.nqp:64  (/priv/rir/Repo/rakudo/install/share/perl6/lib/Perl6/Compiler.moarvm:command_eval)
 from gen/moar/stage2/NQPHLL.nqp:2035  (/priv/rir/Repo/rakudo/install/share/nqp/lib/NQPHLL.moarvm:command_line)
 from gen/moar/rakudo.nqp:126  (/priv/rir/Repo/rakudo/install/share/perl6/runtime/perl6.moarvm:MAIN)
 from gen/moar/rakudo.nqp:1  (/priv/rir/Repo/rakudo/install/share/perl6/runtime/perl6.moarvm:<mainline>)
 from <unknown>:1  (/priv/rir/Repo/rakudo/install/share/perl6/runtime/perl6.moarvm:<main>)
 from <unknown>:1  (/priv/rir/Repo/rakudo/install/share/perl6/runtime/perl6.moarvm:<entry>)

```

* Operating system:  Linux shrew 4.19.0-12-amd64 #1 SMP Debian 4.19.152-1 (2020-10-18) x86_64 GNU/Linux
* Compiler version:  This is Rakudo version 2020.07-66-g2db92e04f built on MoarVM version 2020.07-16-g03d3e43fa
implementing Raku 6.d.

**Comment by vrurg** [comment]
@rakool I'd like to ask you to provide full example of the failing code.

**Comment by jnthn** [comment]
I suspect it's not about attributes, but about this line:

> my $c = ::($o.^name);

It can only look up things that are in scope. If it doesn't find the thing, then it returns `Failure`.

**Comment by rakool** [comment]
I  will be happy to put up all code.  Is there a better way than inline here in separate blocks?  (And @jnthn seems to have touched the heart of the matter.)
```
my $c = ::($o.^name); 
```
@jnthn  I see you are right.  With Raku thus far, I have been naive about **how**  I have stringified to check code.  Here I used `say "::($o.^name())"`  when `(::($o.^name)).gist.say` would be more helpful.

I want to have as little as possible in my derived classes and a SPoT for this functionality.  I guess my best option is to put all my child classes in one unit along with the single helper routine.

So I think  this issue has little merit aside from my epiphanies.   Thank you.
